.buildkite/pipelines/code_coverage/daily.yml

```diff
@@ -0,0 +1,46 @@
+steps:
+  - command: .buildkite/scripts/lifecycle/pre_build.sh
+    label: Pre-Build
+    timeout_in_minutes: 10
+    agents:
+      queue: kibana-default
+
+  - wait
+  
+  - command: .buildkite/scripts/steps/code_coverage/jest.sh
+    label: 'Jest Tests'
+    parallelism: 8
+    agents:
+      queue: ci-group-4d
+    timeout_in_minutes: 120
+    key: jest
+    
+  - command: .buildkite/scripts/steps/code_coverage/oss_cigroup.sh
+    label: 'OSS CI Group'
+    parallelism: 11
+    agents:
+      queue: c2-16
+    timeout_in_minutes: 120
+    key: oss-cigroup
+
+  - command: .buildkite/scripts/steps/code_coverage/xpack_cigroup.sh
+    label: 'Default CI Group'
+    parallelism: 29
+    agents:
+      queue: c2-16
```

Comment:
c2-16s are pretty big machines compared to what we normally run functional tests on, and you're spinning a lot of them up... are you sure you need that much CPU?

Reply:
there was a problem with building plugins with smaller machines. https://github.com/elastic/kibana/blob/main/.buildkite/scripts/steps/code_coverage/xpack_cigroup.sh#L8

Since we run tests from source to collect code coverage from client & server, I moved building plugins in each thread. Maybe it is possible to build it once and later download and re-use with source code?

Reply:
This happens during the build step: https://github.com/elastic/kibana/blob/main/.buildkite/scripts/post_build_kibana_plugins.sh#L5

And they get downloaded here and extracted to the kibana repo directory: https://github.com/elastic/kibana/blob/main/.buildkite/scripts/download_build_artifacts.sh#L11-L18

Maybe that will suffice?